python/rmm/_lib/memory_resource.pyx

```diff
@@ -312,11 +313,12 @@ cpdef void _initialize(
 
     if pool_allocator:
         typ = PoolMemoryResource
-        if initial_pool_size is None:
-            args = (upstream(),)
-        else:
-            args = (upstream(), initial_pool_size)
-
+        args = [upstream()]
+        if initial_pool_size is not None:
+            args.append(initial_pool_size)
+        if maximum_pool_size is not None:
+            args.append(maximum_pool_size)
```

Comment:
This looks fishy, if depending on the value of `initial_pool_size`, the argument for `maximum_pool_size` can be at position 1 or 2 in the list.

Reply:
Ah nice catch - oversight on my part